straxen/plugins/multi_scatter.py

```diff
@@ -0,0 +1,325 @@
+        if isinstance(self.se_gain, dict):
+            seg = self.se_gain
+        else:
+            seg = {key: self.se_gain for key in regions}
+
+        if isinstance(self.avg_se_gain, dict):
+            avg_seg = self.avg_se_gain
+        else:
+            avg_seg = {key: self.avg_se_gain for key in regions}
+
+        if isinstance(self.rel_extraction_eff, dict):
+            ee = self.rel_extraction_eff
+        else:
+            ee = {key: self.rel_extraction_eff for key in regions}
+
+        # now can start doing corrections
+
+            # S2(x,y) corrections use the observed S2 positions
+        s2_positions = np.vstack([peaks[f'x'], peaks[f'y']]).T
+
+        # corrected s2 with s2 xy map only, i.e. no elife correction
+        # this is for s2-only events which don't have drift time info
+
+        cs2_top_xycorr = (peaks[f'area']
+                          * peaks[f'area_fraction_top']
+                          / self.s2_xy_map(s2_positions, map_name=s2_top_map_name))
+        cs2_bottom_xycorr = (peaks[f'area']
+                             * (1 - peaks[f'area_fraction_top'])
+                             / self.s2_xy_map(s2_positions, map_name=s2_bottom_map_name))
+
+        # For electron lifetime corrections to the S2s,
+        # use drift time computed using the main S1.
+        
+        elife_correction = np.exp(peaks[f'drift_time'] / self.elife)
+        result[f"cs2_wo_timecorr"] = ((cs2_top_xycorr + cs2_bottom_xycorr) * elife_correction)
+
+        for partition, func in regions.items():
+            # partitioned SE and EE
+            partition_mask = func(peaks[f'x'], peaks[f'y'])
+
+            # Correct for SEgain and extraction efficiency
+            seg_ee_corr = seg[partition]/avg_seg[partition]*ee[partition]
+
+            # note that these are already masked!
+            cs2_top_wo_elifecorr = cs2_top_xycorr[partition_mask] / seg_ee_corr
+            cs2_bottom_wo_elifecorr = cs2_bottom_xycorr[partition_mask] / seg_ee_corr
+
+            result[f"cs2_wo_elifecorr"][partition_mask] = cs2_top_wo_elifecorr + cs2_bottom_wo_elifecorr
+
+            # cs2aft doesn't need elife/time corrections as they cancel
+            result[f"cs2_area_fraction_top"][partition_mask] = cs2_top_wo_elifecorr / (cs2_top_wo_elifecorr + cs2_bottom_wo_elifecorr)
+
+            result[f"cs2"][partition_mask] = result[f"cs2_wo_elifecorr"][partition_mask] * elife_correction[partition_mask]
+            result[f"cs2_bottom"][partition_mask] = cs2_bottom_wo_elifecorr * elife_correction[partition_mask]
+        result[f"cs2_wo_timecorr"][peaks["type"]!=2] = np.nan
+        result[f"cs2_wo_elifecorr"][peaks["type"]!=2] = np.nan
+        result[f"cs2_area_fraction_top"][peaks["type"]!=2] = np.nan
+        result[f"cs2"][peaks["type"]!=2] = np.nan
+        result[f"cs2_bottom"][peaks["type"]!=2] = np.nan
+        return result
+   
+
+@export
+@strax.takes_config(
+    strax.Option(
```

Comment:
Can you make this a URLConfig?

Reply:
Sorry I am not sure what you want to make in to a URL config